src/server/scripts/Northrend/VioletHold/boss_xevozz.cpp

```diff
@@ -225,72 +243,132 @@ class npc_ethereal_sphere : public CreatureScript
     {
         npc_ethereal_sphereAI(Creature* creature) : ScriptedAI(creature)
         {
-            Initialize();
-            instance   = creature->GetInstanceScript();
+            instance = creature->GetInstanceScript();
         }
 
-        void Initialize()
+        void Reset() override
         {
-            uiSummonPlayers_Timer = urand(33000, 35000);
-            uiRangeCheck_Timer = 1000;
+            events.Reset();
+            DoCast(SPELL_POWER_BALL_VISUAL);
+            DoCast(SPELL_POWER_BALL_DAMAGE_TRIGGER);
+            arcanePower = false;
+            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE);
+            me->setFaction(16);
+
+            events.ScheduleEvent(EVENT_DESPAWN_SPHERE, 40000);
+            events.ScheduleEvent(EVENT_RANGE_CHECK, 1000);
         }
 
-        InstanceScript* instance;
-
-        uint32 uiSummonPlayers_Timer;
-        uint32 uiRangeCheck_Timer;
-
-        void Reset() override
+        void DoAction(int32 action) override
         {
-            Initialize();
+            DoCast(SPELL_SUMMON_PLAYERS);
         }
 
         void UpdateAI(uint32 diff) override
         {
-            if (!UpdateVictim())
-                return;
+            events.Update(diff);
 
-            if (!me->HasAura(SPELL_POWER_BALL_VISUAL))
-                DoCast(me, SPELL_POWER_BALL_VISUAL);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
 
-            if (uiRangeCheck_Timer < diff)
+            switch (uint32 eventId = events.ExecuteEvent())
```

Comment:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_xevozz.cpp: In member function ‘virtual void npc_ethereal_sphere::npc_ethereal_sphereAI::UpdateAI(uint32)’:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_xevozz.cpp:289:28: warning: unused variable ‘eventId’ [-Wunused-variable]
             switch (uint32 eventId = events.ExecuteEvent())
                            ^